_test/test_sqw_class/test_unary_ops.m

```diff
@@ -0,0 +1,141 @@
+classdef test_unary_ops < TestCase
+
+methods
+
+    function obj = test_unary_ops(~)
+        obj = obj@TestCase('test_unary_ops');
+    end
+
+    function test_all_functions_are_defined(obj)
+        % the unary operation and the range the data it acts on should take
+        unary_ops = {
+            @acos, [0, 1], ...
+            @acosh, [1, 3], ...
+            @acot, [0, 1], ...
+            @acoth, [10, 15], ...
+            @acsc, [1, 3], ...
+            @acsch, [1, 3], ...
+            @asec, [1.5, 3], ...
+            @asech, [0, 1], ...
+            @asin, [0, 1], ...
+            @asinh, [1, 3], ...
+            @atan, [0, 1], ...
+            @atanh, [0, 0.5], ...
+            @cos, [0, 1], ...
+            @cosh, [0, 1], ...
+            @cot, [0, 1], ...
+            @coth, [1.5, 3], ...
+            @csc, [0.5, 2.5], ...
+            @csch, [1, 3], ...
+            @exp, [0, 1], ...
+            @log, [1, 3], ...
+            @log10, [1, 3], ...
+            @sec, [2, 4], ...
+            @sech, [0, 1.4], ...
+            @sin, [0, 3], ...
+            @sinh, [0, 3], ...
+            @sqrt, [0, 3], ...
+            @tan, [0, 1], ...
+            @tanh [0, 3], ...
+        };
+
+        % For each unary operator, perform the operation on some data
+        num_pix = 7;
+        for i = 1:numel(unary_ops)/2
+            unary_op = unary_ops{2*i - 1};
+            data_range = unary_ops{2*i};
+
+            data = obj.get_random_data_in_range( ...
+                PixelData.DEFAULT_NUM_PIX_FIELDS, num_pix, data_range);
+
+            sqw_obj = sqw();
+            sqw_obj.data.pix = PixelData(data);
+            sqw_obj.data.npix = [num_pix];
+            sqw_obj.data.s = -99;
+            sqw_obj.data.e = -99;
+
+            % exception thrown if method not implemented
+            result = unary_op(sqw_obj);
+
+            % confirm the raw pixel data has changed
+            assertFalse(equal_to_tol(result.data.pix.signal, sqw_obj.data.pix.signal))
+            
+            % confirm the (previously unset) image data (s, e) has been
+            % calculated correctly from the updated raw pixels
+            assertEqualToTol(result.data.s, mean(result.data.pix.signal));
+            assertEqualToTol(result.data.e, sum(result.data.pix.variance)./num_pix^2);
+        end
+    end
+
+
+    function test_unary_op_updates_image_signal_and_error_if_no_pixeldata(~)
+
+        sqw_obj = sqw();
+        sqw_obj.data.pix = PixelData();
+        sqw_obj.data.s = 2;
+        sqw_obj.data.e = 1.5;
+
+        result = log10(sqw_obj);
+
+        % explicit calculation test
+        expected_signal = log10(sqw_obj.data.s);
+        expected_var = sqw_obj.data.e./(sqw_obj.data.s*log(10)).^2;
+        
+        assertEqualToTol(result.data.s, expected_signal);
+        assertEqualToTol(result.data.e, expected_var);
+    end
+
+    function test_unary_op_updates_image_signal_and_error(obj)
```

Comment:
Would this test provide better coverage if we used arrays for the image rather than a scalar?

Reply:
I chose scalars here as the "simplest" data needed to confirm the two things I'm testing here i) the image data is updated and ii) the correct update is applied.

The (extensive) testing in the `PixelData` class of the unary operations proves that they perform the correct calculations to raw pixel and image data.